hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -242,11 +287,12 @@ class ExpressionPayload(record: GenericRecord,
    *
    * @return
    */
-  private def joinRecord(sourceRecord: IndexedRecord, targetRecord: IndexedRecord): IndexedRecord = {
+  private def joinRecord(sourceRecord: IndexedRecord, targetRecord: IndexedRecord): GenericRecord = {
     val leftSchema = sourceRecord.getSchema
     val joinSchema = getMergedSchema(leftSchema, targetRecord.getSchema)
 
-    val values = new ArrayBuffer[AnyRef]()
+    // TODO rebase onto JoinRecord
```

Comment:
still needed?

Reply:
Yeah, leaving these for subsequent follow-up